src/test/java/com/googlecode/download/maven/plugin/internal/WGetTest.java

```diff
@@ -1,76 +1,279 @@
-                    Matchers.emptyArray());
+            WGet firstMojo = createMojo(mojo ->
+                    setVariableValueToObject(mojo, "uri", outputFile.toURI()));
+            WGet secondMojo = createMojo(mojo ->
+                    setVariableValueToObject(mojo, "uri", secondOutputFile.toURI()));
+
+            CountDownLatch firstMojoStarted = new CountDownLatch(1), secondMojoFinished = new CountDownLatch(1);
+            Arrays.asList(CompletableFuture.runAsync(() -> {
+                        try (MockedStatic<HttpClientBuilder> httpClientBuilder = mockStatic(HttpClientBuilder.class)) {
+                            HttpClientBuilder clientBuilder = mockClientBuilder(() -> {
+                                firstMojoStarted.countDown();
+                                try {
+                                    // waiting till we're sure the second mojo has finished
+                                    assert secondMojoFinished.await(1, SECONDS);
+                                } catch (InterruptedException e) {
+                                    throw new RuntimeException(e);
+                                }
+                                return new ByteArrayInputStream("foo\n".getBytes());
+                            });
+                            httpClientBuilder.when(HttpClientBuilder::create).thenReturn(clientBuilder);
+                            firstMojo.execute();
+                        } catch (Exception e) {
+                            throw new RuntimeException(e);
+                        }
+                    }),
+                    CompletableFuture.runAsync(() -> {
+                        try (MockedStatic<HttpClientBuilder> httpClientBuilder = mockStatic(HttpClientBuilder.class)) {
+                            HttpClientBuilder clientBuilder = mockClientBuilder(() ->
+                                    new ByteArrayInputStream("bar\n".getBytes()));
+                            httpClientBuilder.when(HttpClientBuilder::create).thenReturn(clientBuilder);
+                            // waiting till we're sure the first mojo has started
+                            assert firstMojoStarted.await(1, SECONDS);
+                            secondMojo.execute();
+                            secondMojoFinished.countDown();
+                        } catch (Exception e) {
+                            throw new RuntimeException(e);
+                        }
+                    })).forEach(CompletableFuture::join);
+
+            // cache should contain both entries
+            @SuppressWarnings("unchecked")
+            Map<URI, String> index = (Map<URI, String>) new ObjectInputStream(Files.newInputStream(
+                    new File(cacheDirectory, "index.ser").toPath())).readObject();
+            assertThat(index.entrySet(), hasSize(2));
+            assertThat(index.keySet(), containsInAnyOrder(outputFile.toURI(), secondOutputFile.toURI()));
```

Comment:
@ajarmoniuk it worth to verify that cached files are there and have proper content as well

Reply:
Implemented. Please check.